src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -225,17 +222,8 @@ private void validateJsonSchemaConstraints(final JSONObject schema) throws Inval
         }
     }
 
-    private void validatePartitionKeys(final Optional<Schema> schemaO, final EventTypeBase eventType)
+    private void validatePartitionKeys(final Schema schema, final EventTypeBase eventType)
             throws InvalidEventTypeException, JSONException, SchemaException {
-        if (!featureToggleService.isFeatureEnabled(CHECK_PARTITIONS_KEYS)) {
```

Comment:
This feature needs to be released. Right now, we do not check if a given partition key exists in the schema which could lead to NO-OP topics: a topic that expects events to contain a given key when partitioning but this same key is not allowed because it's not in the schema definition.

I have no clue why it wasn't turned on. I can keep it but then we need to turn it on altogether.